docs/writing_rules/examples.md

```diff
@@ -0,0 +1,232 @@
+
```

Comment:
I really like the tone, style, and examples in this, awesome work! 🎉 🕺 

Could you make a quick index at the top, like https://github.com/returntocorp/semgrep/blob/develop/docs/pattern-features.md, that makes it easy to get a glimpse into what's contained and jump to a specific example?